internal/run/run.go

```diff
@@ -130,51 +131,35 @@ func (w *Workflow) RunInner(ctx context.Context) error {
 		return fmt.Errorf("cannot specify both a target and a source")
 	}
 
-	if w.Target == "all" {
-		if w.SetVersion != "" && len(w.workflow.Targets) > 1 {
-			return fmt.Errorf("cannot manually apply a version when more than one target is specified ")
-		}
-
-		for t := range w.workflow.Targets {
-			sourceRes, targetRes, err := w.runTarget(ctx, t)
-			if sourceRes != nil {
-				w.SourceResults[sourceRes.Source] = sourceRes
-			}
-			w.TargetResults[t] = targetRes
+	if w.SetVersion != "" && len(w.workflow.Targets) > 1 {
+		return fmt.Errorf("cannot manually apply a version when more than one target is specified ")
+	}
 
-			if err != nil {
-				return err
-			}
+	sourceIDs := []string{w.Source}
+	if w.Source == "all" {
+		sourceIDs = lo.Keys(w.workflow.Sources)
+	}
+	targetIDs := []string{w.Target}
+	if w.Target == "all" {
+		targetIDs = lo.Keys(w.workflow.Targets)
+	}
 
+	for _, sourceID := range sourceIDs {
+		if sourceID == "" {
+			continue
 		}
-	} else if w.Source == "all" {
-		for id := range w.workflow.Sources {
-			_, sourceRes, err := w.RunSource(ctx, w.RootStep, id, "")
-			if sourceRes != nil {
-				w.SourceResults[sourceRes.Source] = sourceRes
-			}
-			if err != nil {
-				return err
-			}
+		if _, ok := w.workflow.Sources[sourceID]; !ok {
+			return fmt.Errorf("source %s not found", sourceID)
 		}
-	} else if w.Target != "" {
-		if _, ok := w.workflow.Targets[w.Target]; !ok {
-			return fmt.Errorf("target %s not found", w.Target)
-		}
-
-		sourceRes, targetRes, err := w.runTarget(ctx, w.Target)
-		if sourceRes != nil {
-			w.SourceResults[sourceRes.Source] = sourceRes
-		}
-		w.TargetResults[w.Target] = targetRes
-
+		_, _, err := w.RunSource(ctx, w.RootStep, sourceID, "")
```

Comment:
Strange that we are capturing sourceRes for Targets but not Sources

Reply:
you mean how it was before?

Reply:
It vaguely looks like we used to capture sourceRes for `w.Source == "all"`, but now don't? Might effect a source only run. Just doing a drive-by inspection though and could be wrong.

![Screenshot 2024-09-09 at 19 16 24](https://github.com/user-attachments/assets/7c1c5900-c843-44da-bc76-7b6d34af9d75)

Reply:
sorry there was a mistake in my refactor should be fixed now, I think these comments are out of context. We capture using defer in `RunSource` and `runTarget`